### DISCUSSION TITLE
База данный режима

### DISCUSSION THREAD

**Discussion by SobakaBarabaka74** [Sokol-Games-GmbH/PixelCombats.PublicApi#238] [Ideas]
**Идея в том чтобы сделать общую базу данных для всех режимов, которая будет действовать на всех серверах с данным режимом одинаково.** Значения будут меняться в рантайме ( тоесть если значение изменено на 1 из серверов, на других тоже изменится). Например это будет реализовано в файле database.json ( который будет естественно не обязателен).
```json
{
    "Properties": {
        "Client": [
             {
                 "Name": "Scores", 
                 "Default": 0
             }
         ], 
         "GameMode": [
              {
                  "Name": "LastPlayerId",
                  "Default": "0"
              }
         ]
    }
}
```
**Пояснение.**
*Properties* - это список всех пропов которые будут сохранены.
*Client* - Это значения клиента (игрока) которые будут сохранены. 
*GameMode* - Это значения на режиме, которые имеют контекст комнаты.

Внутри каждого из массивов есть структура, в которой есть 2 свойства:
*Name* - Имя свойства. 
*Default* - Дефолтное значение, тоесть если значение является null, то грузить из Default.

**Доступ из скрипта**
База данных будет доступна в скриптах, а так же в плагинах. Изменятся оно будет так же как и обычное проперти.
```js
// Меняет значение в базе данных по имени LastPlayerId
Properties.GetContext().Get("LastPlayerId").Value = "Test";
// тоже самое с клиентом, только сменить контекст проперти
```
Думаю я описал всё что хотел давольно просто и понятно, могу сделать Api структур по надобности.

**Comment by kkohno** [comment]
Да, такая идея у нас есть, ее сделаем. А что если в какомто режиме читер сменит опцию всего режима? Какаято защита нужна я так понимаю... Если можете сделать апи структур то хорошобы, посмотрим, примем во внимание

**Reply by SobakaBarabaka74** [reply]
Ну тут защита нужна вообще от читеров (в общем контексте). Вы можете как то сдалать защиту эту? Лучше потратите этот месяц-2, но зато потом спокойно делать обновы

**Reply by SobakaBarabaka74** [reply]
Сделал пулл реквест
